source/_posts/react-localstorage.md

```diff
@@ -0,0 +1,382 @@
+
+Before we need to configure that encrypted storage, for that create a new js file like this:
+
+```js
+import { EncryptStorage } from 'encrypt-storage'
+
+const encryptedLocalStorage = new EncryptStorage(
+  'SET_YOUR_SECRET_KEY',
+  {
+    // Keys used by this library will have this prefix
+    // e.g.: 'enc' + ':' + 'input-data' = 'enc:input-data' as key name
+    prefix:'enc',
+    // Encryption algorithm type
+    encAlgorithm: 'AES',
+    // Storage type (localStorage and sessionStorage are supported)
+    storageType: 'localStorage'
+  }
+);
+
+export {
+  encryptedLocalStorage
+}
+```
+
+The `encryptedLocalStorage` is essentially equal to `localStorage`, but it has additional implementations to encrypt the store and be used as `localStorage` object.
+
+We already know how to create a React custom hook with `localStorage`. We need to create another similar using the `encryptedLocalStorage` store object. Let's update the `localstorage.js` file (created earlier).
+
+```js
+import { useEffect, useState } from "react"
+import { encryptedLocalStorage } from "../utils/secureLocalStorage"
+
+
+const decode = (value) => {
+  return JSON.stringify(value)
+}
+
+const encode = (value) => {
+  return JSON.parse(value)
+}
+
+// useLocalStorage hook
+const useLocalStorage = (key, defaultState) => {
+  const [value, setValue] = useState(
+    encode(localStorage.getItem(key)||null) || defaultState
+  )
+
+  useEffect(() => {
+    localStorage.setItem(key, decode(value))
+  },  [value])
+
+  return [value, setValue]
+}
+
+// New encrypted localStorage
+const useSecureLocalStorage = (key, defaultState) => {
+  const [value, setValue] = useState(
+    encode(encryptedLocalStorage.getItem(key)||null) || defaultState
+  )
+
+  useEffect(() => {
+    encryptedLocalStorage.setItem(key, decode(value))
+  },  [value])
+
+  return [value, setValue]
+}
+
+export {
+  useLocalStorage,
+  useSecureLocalStorage
+}
+```
+
+So, it's time to use the `useSecureLocalStorage` hook.
+
+```js
+import { useSecureLocalStorage } from "../hooks/localstorage";
+
+export default function SecureHookMode(){
+  const [counter, setCounter] = useSecureLocalStorage('counter', 0);
+  const [secondCounter, setSecondCounter] = useSecureLocalStorage('secondCounter', 0);
+  
+  return (
+    <div className="card">
+      <h3>Secure Hook Example</h3>
+      <p>Click the button many times and try to refresh the page</p>
+      <br />
+
+      <button onClick={() => setCounter(counter + 1)}>
+        Button 1: encrypted count is {counter}
+      </button>
+      <br /><br />
+      <button onClick={() => setSecondCounter(secondCounter + 1)}>
+        Button 2: encrypted count is {secondCounter}
+      </button>
+    </div>
+  )
+}
+```
+
+Use the `developer tools` go to Application>Local Storage tool and check the `localStorage`. The stored data is currently encrypted 😎 and is prefixed with our selected prefix (`enc` in this case).
```

Comment:
`developer tools` - is that some syntax to be highlighted? :D

Reply:
Done :D